src/view/MapVisualizationView.java

```diff
@@ -1,2 +1,240 @@
-package view;public class MapVisualizationView {
+package view;
+
+import interface_adapter.search.SearchState;
+import interface_adapter.visualize.VisualizeController;
+import interface_adapter.visualize.VisualizeState;
+import interface_adapter.visualize.VisualizeViewModel;
+import org.openstreetmap.gui.jmapviewer.*;
+import org.openstreetmap.gui.jmapviewer.tilesources.BingAerialTileSource;
+import org.openstreetmap.gui.jmapviewer.tilesources.OsmTileSource;
+import resources.map.MapFont;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+
+public class MapVisualizationView extends JFrame implements PropertyChangeListener, ActionListener {
+    /* Internal strings */
+    public final String viewName = "visualize";
+    private final VisualizeViewModel visualizeViewModel;
+//    private final VisualizeController visualizeController;
+
+    /*
+     The controller is ignored since no entity/external data needs to be retrieved or modified once this view
+     is displayed + this is a popup. This will change should a refresh button be implemented.
+
+     In regard to a refresh button, query data would need to be resent again through the APi (i.e. call
+     ShowIncomingVehicles use case files, then route the output through MapVisualization use case files)
+     so not too sure about how to handle this. Will discuss with team.
+    */
+
+    /* JSwing elements */
+    private final JMapViewer map;
+
+    JPanel panel;
+    JPanel panelTop;
+    JPanel helpPanel;
+
+    private Coordinate c(double lat, double lon) {
+        return new Coordinate(lat, lon);
+    }
+
+    private Coordinate c(String lat, String lon) {
+        return new Coordinate(Double.parseDouble(lat), Double.parseDouble(lon));
+    }
+
+    // denote default style for markers
+    // TODO: Depending on parent line info, change the color of the inner point
+    private final Style defaultStyle = new Style(
+            Color.cyan, new Color(245, 128, 37), new BasicStroke(10), new MapFont().getFont());
+
+//    public MapVisualizationView(VisualizeViewModel visualizeViewModel, VisualizeController visualizeController) {
+public MapVisualizationView(VisualizeViewModel visualizeViewModel) {
```

Comment:
The indenting of this method is a bit off, considering refactoring how this method is formatted for code readability